### PR TITLE
desktop-gui: Add toggle buttons to root specs to expand/collapse all their subfolders

### LINKED ISSUE #6282: Create a toggle to close/open all folders in Tests list
**Current Behavior:**
Currently there seems to be no way to have all your folders defaulted to close when you open up Cypress.

The results is a messy UI that isn't as clean when you try and organize your many Cypress tests into folders.

**Desired Behavior:**
Have a setting option to default your folders to close on Cypress open that way you can navigate a cleaner interface when you organize your multiple tests into their appropriate folders.

Screen Shot 2020-01-30 at 4 46 20 PM

### PR DESCRIPTION
**Pull request by martinsik**
- Closes #6282

### User facing changelog

There is now "collapse all" and "expand all" buttons in the Test Runner's 'Tests' tab to collapse or expand all folders within the `integrationFolder` .

### How has the user experience changed?

This PR adds two buttons to UI next to each root spec name. Clicking "Collapse all" will only collapse subfolders and will keep the root spec expanded.

#### Collapse All /Expand All links not shown when no folders

Screen Shot 2020-02-27 at 1 55 47 PM

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by martinsik** [comment]
I'm not sure why there are two failing tests in `commands/xhr_spec.coffee` and `commands/angular_spec.coffee`.

File: packages/desktop-gui/cypress/integration/specs_list_spec.js [hunk 1]
```diff
@@ -279,6 +279,61 @@ describe('Specs List', function () {
...
+        this.openProject.resolve(this.config)
+      })
+
+      it('collapsing root spec will keep root itself expanded', () => {
+        cy.get('.level-0 .folder-name').find('a:first').click({ multiple: true })
+        cy.get('.folder.folder-collapsed').should('have.length', 3)
+        cy.get('.folder.folder-expanded').should('have.length', 2)
+      })
+
+      it('collapses all children folders', () => {
```
Comment (jennifer-shehane) [inline packages/desktop-gui/cypress/integration/specs_list_spec.js]: This is an interesting way to test this 👍

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
I think the `collapse all | expand all` links should be removed if there are no folders to expand/collapse, otherwise if just seems like the links don't do anything and is more clutter for users that don't have a larger test suite.

**Review by jennifer-shehane — DISMISSED** [review]
@martinsik Some linting errors need fixing. You can run `yarn lint-fix` in the root of the project to automatically fix these before committing.

**Comment by martinsik** [comment]
@jennifer-shehane I tried running `yarn lint-fix` but it went through the whole project.

The lint check is now throwing:
```
/root/cypress/packages/desktop-gui/src/specs/specs-store.js
  146:16  error  'child' is defined but never used  no-unused-vars
```
But that's  on this line which is weird.
https://github.com/cypress-io/cypress/pull/6541/files#diff-2bf56b7e7d242f95dffdd2309a9f2948R145-R152

It might be related to this: https://github.com/eslint/eslint/issues/12165

**Review by chrisbreiding — DISMISSED** [review]

**Comment by jennifer-shehane** [comment]
Final cosmetic change after asking team - wanted to move the 'collapse/expand' buttons to the right to clean up the UI clutter some.

**Review by jennifer-shehane — APPROVED** [review]

**Comment by gunslingor** [comment]
What's the status of this? We have thousands of cy files in many folders, sure would be nice to collapse so we can see structure. An alternative would be to turn it into a table with path field and all column filter.

**Comment by martinsik** [comment]
@gunslingor This PR was merged a while ago into the old UI. I guess it was rewritten from scratch or redesigned heavily, so even though this PR was merged it's probably not relevant any more.

**Pull request merged**